--- a/drivers/net/ethernet/airoha/airoha_eth.c
+++ b/drivers/net/ethernet/airoha/airoha_eth.c
@@ -3087,6 +3087,8 @@ static int airoha_qdma_set_trtcm_param(s
 				       u32 addr, enum trtcm_param_type param,
 				       enum trtcm_mode_type mode, u32 val)
 {
+	u32 value_lo = 0, value_hi = 0;
+	int retry_num = 100;
 	u32 idx = QDMA_METER_IDX(channel), group = QDMA_METER_GROUP(channel);
 	u32 config = TRTCM_PARAM_RW_MASK |
 		     FIELD_PREP(TRTCM_PARAM_TYPE_MASK, param) |
@@ -3094,8 +3096,15 @@ static int airoha_qdma_set_trtcm_param(s
 		     FIELD_PREP(TRTCM_PARAM_INDEX_MASK, idx) |
 		     FIELD_PREP(TRTCM_PARAM_RATE_TYPE_MASK, mode);
 
-	airoha_qdma_wr(qdma, REG_TRTCM_DATA_LOW(addr), val);
-	airoha_qdma_wr(qdma, REG_TRTCM_CFG_PARAM(addr), config);
+	do {
+		airoha_qdma_wr(qdma, REG_TRTCM_DATA_LOW(addr), val);
+		wmb();
+		airoha_qdma_wr(qdma, REG_TRTCM_CFG_PARAM(addr), config);
+		airoha_qdma_get_trtcm_param(qdma, channel, addr, param, mode,
+					    &value_lo, &value_hi);
+		if (val == value_lo)
+			break;
+	} while (--retry_num);
 
 	return read_poll_timeout(airoha_qdma_rr, val,
 				 val & TRTCM_PARAM_RW_DONE_MASK,
@@ -4250,8 +4259,10 @@ static int airoha_en7581_get_dev_from_sp
 
 	*dev = 0;
 	switch (sport) {
-	case 0x10 ... 0x14:
-		*port = 0; /* GDM1 */
+	case 0x0 ... 0x1:	/* pingpong */
+	case 0x7:		/* npu pingpong */
+	case 0x10 ... 0x14:	/* GDM1 */
+		*port = 0;
 		break;
 	case 0x2 ... 0x4:
 		*port = sport - 1;
@@ -4331,8 +4342,10 @@ static int airoha_an7583_get_dev_from_sp
 
 	*dev = 0;
 	switch (sport) {
-	case 0x10 ... 0x14:
-		*port = 0; /* GDM1 */
+	case 0x0 ... 0x1:	/* pingpong */
+	case 0x7:		/* npu pingpong */
+	case 0x10 ... 0x14:	/* GDM1 */
+		*port = 0;
 		break;
 	case 0x2 ... 0x4:
 		*port = sport - 1;
--- a/drivers/net/ethernet/airoha/airoha_eth.h
+++ b/drivers/net/ethernet/airoha/airoha_eth.h
@@ -515,7 +515,7 @@ struct airoha_wdma_info {
 };
 
 /* RX queue to IRQ mapping: BIT(q) in IRQ(n) */
-#define RX_IRQ0_BANK_PIN_MASK			0x839f
+#define RX_IRQ0_BANK_PIN_MASK			0xbb9f
 #define RX_IRQ1_BANK_PIN_MASK			0x7fe00000
 #define RX_IRQ2_BANK_PIN_MASK			0x20
 #define RX_IRQ3_BANK_PIN_MASK			0x40
