--- a/include/net/sock.h
+++ b/include/net/sock.h
@@ -3082,7 +3082,7 @@ void sk_get_meminfo(const struct sock *s
 #define _SK_MEM_PACKETS		256
 #define _SK_MEM_OVERHEAD	SKB_TRUESIZE(256)
 #define SK_WMEM_DEFAULT		(_SK_MEM_OVERHEAD * _SK_MEM_PACKETS)
-#define SK_RMEM_DEFAULT		(_SK_MEM_OVERHEAD * _SK_MEM_PACKETS)
+#define SK_RMEM_DEFAULT		(1UL << 20)
 
 extern __u32 sysctl_wmem_max;
 extern __u32 sysctl_rmem_max;
